src/briefcase/platforms/linux/appimage.py

```diff
@@ -259,16 +259,33 @@ def run_app(self, app: AppConfig, **kwargs):
         """
         self.logger.info("Starting app...", prefix=app.app_name)
         try:
-            # Start streaming logs for the app.
-            self.logger.info("=" * 75)
-            self.tools.subprocess.run(
+            # Start the app in a way that lets us stream the logs
+            log_popen = self.tools.subprocess.Popen(
                 [os.fsdecode(self.binary_path(app))],
-                check=True,
                 cwd=self.tools.home_path,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+                bufsize=1,
             )
+
+            try:
+                # Start streaming logs for the app.
+                self.logger.info(
+                    "Following log output (type CTRL-C to stop log)...",
+                    prefix=app.app_name,
+                )
+                self.logger.info("=" * 75)
+                self.tools.subprocess.stream_output(
+                    app.app_name,
+                    log_popen,
+                    stop_func=lambda: log_popen.poll() is not None,
+                )
```

Comment:
curious....did you find the `stop_func` was necessary for log streaming to exit properly?

I'd expect checking `poll()` is functionally equivalent to the log streamer exiting when `readline()` returns an empty string....since that'll happen once the PIPE is closed...and that'll happen when the underlying process exits.

Reply:
Good point - this was mostly done for consistency with the iOS/macOS/etc backends, but you're right - as we're streaming the actual process, and the actual process will exit, there's no need to poll here.